app/api/v1/api.py

```diff
@@ -789,12 +782,14 @@ async def api_get_replay(
             "True",  # TODO: ??
         ).encode(),
     ).hexdigest()
-
     # create a buffer to construct the replay output
     replay_data = bytearray()
-
     # pack first section of headers.
-    replay_data += struct.pack("<Bi", row["mode"], 20200207)  # TODO: osuver
+    replay_data += struct.pack(
+        "<Bi",
+        as_vanilla(row["mode"]),
```

Comment:
this doesn't work, should be `GameMode(row["mode"]).as_vanilla,`

Reply:
huh?

Reply:
oh wow sorry, i use a different version of the api which doesn't require this

Reply:
> this doesn't work, should be `GameMode(row["mode"]).as_vanilla,`

why not `row["mode"] % 4` instead?

Reply:
then you would have duplicated logic and if by any chance gamemodes get changed you would have random places to change this